torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2155,6 +2156,35 @@ void ProcessGroupNCCL::checkAndSetRemoteError() {
   }
 }
 
+// NCCL recommends to evenly distribute ncclUniqueIds accross the ranks
+// https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/usage/communicators.html#init-rank-config
+// Let’s consider an example where:
+// nRanks = 10 (total ranks),
+// nIds = 3 (roots),
+// rmr = 10 % 3 = 1 (1 larger group),
+// rpr = 10 / 3 = 3 (base number of ranks per group).
+// rlim = 4
+// Output root:
+// For ranks [0, 1, 2, 3] is 0.
+// For ranks [4, 5, 6] is 1.
+// For ranks [7, 8, 9] is 2.
```

Comment:
Are the output roots 0, 1, 2 absolute rank IDs?
This choice seems different from recommendation:
"For example, if 3 ncclUniqueIds are to be distributed accross 7 NCCL ranks, the first ncclUniqueId will be associated to ranks 0-2, while the others will be associated to ranks 3-4, and 5-6. This function will therefore return true on **rank 0, 3, and 5**, and false otherwise."

Reply:
oh, I think it gives out 0, 3 and 5 in this case..